### ISSUE TITLE
The inference time of the Critic/Reference/Reward model is abnormal

### ISSUE THREAD

**Issue by hua0x522** [OpenRLHF/OpenRLHF#1035]
## Problem
I tested the time of RLHF LLaMA-7B on 8 L40s and found that the inference time of the Critic model / Reference model / Reward model is much shorter than the time for Actor generation.

## Time Info

<img width="1452" alt="Image" src="https://github.com/user-attachments/assets/e44027dd-58af-4785-bb5c-caba8ab1cfc8" />

## Measurement Method

<img width="871" alt="Image" src="https://github.com/user-attachments/assets/eb84d05f-482c-44bc-b39c-14801e9df450" />

## Script
````
#!/bin/bash

set -x

# 生成训练时间戳
TRAIN_TIMESTAMP=$(date +%Y%m%d_%H%M%S)

# 定义基础路径
BASE_DIR="/workspace/OpenRLHF"

# 设置环境变量
export MASTER_ADDR=$(hostname -I | awk '{print $1}')
export MASTER_PORT=29500
export WORLD_SIZE=1
export LOCAL_RANK=0
export RANK=0

# 1. 确保Ray完全停止并清理
ray stop
sleep 1
ray stop
sleep 1

# 清理Ray的临时文件
rm -rf /tmp/ray
rm -rf ~/.ray
sleep 1

# 2. 启动Ray (添加更多配置参数)
ray start --head \
    --node-ip-address ${MASTER_ADDR} \
    --port=6379 \
    --dashboard-port=8265 \
    --num-gpus 8 \
    --num-cpus 56 \
    --object-store-memory 120000000000 \
    --redis-max-memory 5000000000

sleep 3  # 增加等待时间

# 验证Ray状态
ray status || exit 1

# 3. 运行训练脚本
cd ${BASE_DIR}

# 定义要排除的文件模式和环境变量
EXCLUDES='{
    "working_dir": "/workspace/OpenRLHF",
    "excludes": [
        "**/*.pt",
        "**/*.bin", 
        "**/*.pth", 
        "**/*.ckpt", 
        "**/*.safetensors",
        "**/wandb/**",
        "**/__pycache__/**"
    ],
    "env_vars": {
        "NCCL_DEBUG": "INFO",
        "NCCL_IB_DISABLE": "1",
        "NCCL_P2P_DISABLE": "1",
        "NCCL_ASYNC_ERROR_HANDLING": "1",
        "NCCL_BLOCKING_WAIT": "1",
        "NCCL_BUFFSIZE": "2097152",
        "CUDA_DEVICE_ORDER": "PCI_BUS_ID",
        "CUDA_VISIBLE_DEVICES": "0,1,2,3,4,5,6,7",
        "CUDA_LAUNCH_BLOCKING": "1",
        "TORCH_DISTRIBUTED_DEBUG": "DETAIL",
        "TORCH_SHOW_CPP_STACKTRACES": "1",
        "TORCH_CUDA_ARCH_LIST": "8.0",
        "RAY_memory_monitor_refresh_ms": "0",
        "TORCH_DISTRIBUTED_BACKEND": "nccl",
        "NCCL_PROTO": "simple",
        "TORCH_DISTRIBUTED_DEBUG": "INFO"
    }
}'

TRAIN_ARGS=" \
    --actor_num_nodes 1 \
    --actor_num_gpus_per_node 2 \
    --ref_num_nodes 1 \
    --ref_num_gpus_per_node 1 \
    --critic_num_nodes 1 \
    --critic_num_gpus_per_node 2 \
    --reward_num_nodes 1 \
    --reward_num_gpus_per_node 1 \
    --pretrain /dev/shm/llama-2-7b-chat-hf \
    --reward_pretrain /dev/shm/llama-2-7b-chat-hf \
    --save_path /checkpoint/llama-2-7b-chat-hf \
    --micro_train_batch_size 16 \
    --train_batch_size 64 \
    --micro_rollout_batch_size 16 \
    --rollout_batch_size 64 \
    --max_samples 1000 \
    --max_epochs 1 \
    --prompt_max_len 512 \
    --generate_max_len 64 \
    --bf16 \
    --actor_learning_rate 5e-7 \
    --critic_learning_rate 9e-6 \
    --init_kl_coef 0.01 \
    --prompt_data /dev/shm/data.json \
    --input_key prompt \
    --normalize_reward \
    --use_kl_estimator_k3 \
    --adam_offload \
    --gradient_checkpointing \
    --flash_attn \
    --zero_stage 3 \
    --vllm_num_engines 1 \
    --vllm_tensor_parallel_size 2 \
    --load_checkpoint"

ray job submit --address="http://127.0.0.1:8265" \
    --runtime-env-json="${EXCLUDES}" \
    -- python3 -m openrlhf.cli.train_ppo_ray ${TRAIN_ARGS}   

# 4. 等待训练完成并清理
echo "等待训练进程完成..."
sleep 3

# 确保所有进程组被正确销毁
python3 -c "
import torch.distributed as dist
try:
    if dist.is_initialized():
        dist.destroy_process_group()
except:
    pass
"

# 停止Ray
ray stop
sleep 1

# 清理临时文件
rm -rf /tmp/ray
rm -rf ~/.ray

# 清理NCCL临时文件
rm -rf /dev/shm/nccl-*
rm -rf /tmp/nccl-*

# 设置环境变量以避免NCCL警告
export NCCL_ASYNC_ERROR_HANDLING=0
export TORCH_NCCL_ASYNC_ERROR_HANDLING=0

````

**Comment by hijkzzz** [comment]
是的不用 vllm generate 会非常慢
